src/lib.rs

```diff
@@ -235,6 +235,28 @@ impl Surface {
         })
     }
 
+    /// Creates a new instance of this struct, using the provided [`HtmlCanvasElement`](web_sys::HtmlCanvasElement).
+    #[cfg(target_arch = "wasm32")]
+    pub fn from_canvas(canvas: web_sys::HtmlCanvasElement) -> Result<Self, SoftBufferError> {
+        let imple = SurfaceDispatch::Web(web::WebImpl::from_canvas(canvas)?);
+
+        Ok(Self {
+            surface_impl: Box::new(imple),
+        })
+    }
+
+    /// Creates a new instance of this struct, using the provided [`HtmlCanvasElement`](web_sys::OffscreenCanvas).
+    #[cfg(target_arch = "wasm32")]
+    pub fn from_offscreen_canvas(
+        offscreen_canvas: web_sys::OffscreenCanvas,
```

Comment:
I'd rather avoid exposing a public dependency on `web-sys` if at all possible (in case we decide to move to `stdweb` or something). Is there a way of representing these types without `web-sys`?

Reply:
I don't know of a way to avoid that. Even [Winit does it](https://docs.rs/winit/0.28.1/wasm32-unknown-unknown/winit/platform/web/trait.WindowExtWebSys.html#tymethod.canvas).

As far as I know `stdweb` is dead, but if you know of an alternative to the `wasm-bindgen` ecosystem I would be very interested!

Reply:
I forgot to mention, it is actually possible to represent these types without `web-sys`: using `JsValue`.
But this would still require `wasm-bindgen` as a public dependency, which doesn't really help if we want to change to an alternative later.